languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml

```diff
@@ -44227,25 +44227,335 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
       <!-- Created by Tiago F. Santos, Portuguese rule, 2019-08-12 -->
       <url>https://pt.wikipedia.org/wiki/Internet%C3%AAs</url>
       <short>'Internetês'</short>
+      <rule>
+	  <antipattern>
+	    <token regexp='yes'>\d+|&operadores_matematicos;</token>
+        <token>n</token>
+		<example>q = p + n</example>
+      </antipattern>
+      <antipattern>
+        <token>n</token>
+        <token regexp='yes'>\d+|&operadores_matematicos;</token>
+		<example>n = q + r</example>
+      </antipattern>
+	  <antipattern>
+	    <token>valor</token>
+	    <token min='0'>de</token>
+        <token>n</token>
+		<example>Encontre o valor de n.</example>
+      </antipattern>
       <antipattern>
-        <token regexp='yes'>[npq]</token>
-        <token regexp='yes' skip='-1'>\d+|&operadores_matematicos;</token>
+	    <token regexp='yes'>\"|\«|\'|\“|\‘</token>
+        <token>n</token>
+	    <token regexp='yes'>\"|\»|\'|\”|\’</token>
+		  <example>Neste caso "n" representa o número.</example>
+      </antipattern>
+      <pattern>
+        <token regexp='yes'>n|ñ|naum<exception case_sensitive='yes'>N</exception></token>
+      </pattern>
+        <message>Linguagem de Internet. Considere escrever <suggestion>não</suggestion> por extenso. Se quis dizer "n", coloque entre aspas.</message>
+		<example correction="não">Ele <marker>n</marker> me mandou mensagem ontem.</example>
+      </rule>
+      <rule>
+	  <antipattern>
+	    <token regexp='yes'>\d+|&operadores_matematicos;</token>
+        <token>q</token>
+		<example>n = p + q</example>
       </antipattern>
       <antipattern>
+        <token>q</token>
         <token regexp='yes'>\d+|&operadores_matematicos;</token>
-        <token regexp='yes'>\p{L}</token>
+		<example>q = n + r</example>
+      </antipattern>
+	  <antipattern>
+	    <token>valor</token>
+	    <token>de</token>
+        <token>q</token>
+		<example>Encontre o valor de q.</example>
+      </antipattern>
+      <antipattern>
+	    <token regexp='yes'>\"|\«|\'|\“|\‘</token>
+        <token>q</token>
+	    <token regexp='yes'>\"|\»|\'|\”|\’</token>
+		  <example>Neste caso "q" representa a quota.</example>
+      </antipattern>
+      <pattern>
+        <token>q<exception case_sensitive='yes'>Q</exception></token>
+      </pattern>
+        <message>Se for linguagem de Internet, considere escrever <suggestion>que</suggestion> por extenso. Se quis dizer "q", coloque entre aspas.</message>
+		<example correction="que">Ele <marker>q</marker> me mandou mensagem ontem.</example>
+      </rule>
+      <rule>
+	  <antipattern>
+	     <token>/</token>
+	     <token>p</token>
```

Comment:
You should use `spacebefore="no"` here, otherwise `/ p /` (with spaces) would also be matched (not sure if that's bad, but probably it's not intended).

Reply:
Also, it is not super important but as a general advice: try to clean the code up a little more, indentation is kinda all over the place ;-), and it makes it hard to read

Reply:
/ p / with spaces should also be detected, so this should be fine as well.